### PR TITLE
Clarify goal in the DOT DSL blurb and description

### LINKED ISSUE #1537: DOT DSL: Exercise goal is not clear in the description
Tonight I started to do the “DOT DSL” exercise in the Rust track. The goal of the exercise is stated as:

> Write a Domain Specific Language similar to the Graphviz dot language.

This sounded rather difficult to be honest, but interesting, so I read on. I read the example provided about the Dot language and the documentation provided on the Builder pattern. Only after downloading and reading through the tests was I able to grasp what the exercise was trying to get me to do, which is creating structs using the builder pattern. My first impression was that the exercise may have been about trying to parse some text into an AST or something of that sort, but was actually a bit easier and more practical.

After pressing the back button from the description page, I noticed the tags: "builder pattern, derive, modules". After that, everything started to click.

I fear this description may scare some students off, and it's a shame because I really liked the exercise itself after getting to know what it was about.

I hope I'm not being too forward, but here are some concrete steps that could be taken to guide the students a bit more.

1. Change the first phrase from

> Write a Domain Specific Language similar to the Graphviz dot language.

to something like

> Build a data structure that is similar to the Graphviz dot language

2. Add some small sections beneath the links to the materials about the builder pattern, to the Graphviz dot language and documentation on Rust modules

Of course, if this is not in the spirit of exercism, feel free to let me know!
But with some feedback, I'd be more than happy to submit a PR as well.

Thanks for your time.

### PR DESCRIPTION
**Pull request by aburd**
Please let me know if it's too wordy.

A big thanks to everybody that helped me on the issues page!

fixes #1537

### CODE REVIEW

**Comment by SleeplessByte** [comment]
I personally don't like the wording `Create a graph`. You're not necessarily creating a graph. In Ruby, I can solve this exercise without creating _any graph_ or Graph datastructure.

Unclear on what the better wording would be.

**Comment by aburd** [comment]
I'll try to look through other languages to come up with better wording. Is there a place where I can see the tracks this exercise is used?

**Comment by SleeplessByte** [comment]
Apart from rust, here are the others:

https://github.com/exercism/python/tree/master/exercises/dot-dsl
https://github.com/exercism/elixir/tree/master/exercises/dot-dsl
https://github.com/exercism/fsharp/tree/master/exercises/dot-dsl
https://github.com/exercism/csharp/tree/master/exercises/dot-dsl

Note that they create a "graph" data structure **except for elixer**, but I don't think the others are really a DSL or what a DSL in that language is meant to be. 

In ruby, I expect the solution to look like `elixer`, something like:
```ruby
require 'dot-dsl'

graph do
  graph bgcolor: yellow do
    a color: red
    b color: blue
    link a, b, color: green
  end
end
```

Or more verbose, without `method_missing`:

```ruby
require 'dot-dsl'

graph do
  graph bgcolor: yellow do
    node :a, color: red
    node :b color: blue
    link :a, :b, color: green
  end
end
```

**Comment by aburd** [comment]
@SleeplessByte 
Your comment was extremely helpful. Thank you. Looking through each track and your Ruby example really helped me grasp how important the wording is here.

> Note that they create a "graph" data structure **except for elixer**, but I don't think the others are really a DSL or what a DSL in that language is meant to be.

I just want to make sure, but you are saying that in all the languages you are not really creating a DSL **except for elixer**. Because it seems to me that in each language, you are returning a data structure of some kind.

**Comment by SleeplessByte** [comment]
Yeah, so returning the `data` structure is not necessarily bad, but for example (cc @ErikSchierboom), I expected the C# [example](https://github.com/exercism/csharp/blob/master/exercises/dot-dsl/Example.cs) to look more like a _fluent interface_ and not have two `add` functions, which are standard C# method names for graphlike datastructures, which are NOT DSL.

**Comment by ErikSchierboom** [comment]
Well, add is not per se for graphlike collections, but it is for collections (which DSL sort of is). I'm not opposed to doing a fluent interface, as it is quite nice to have an exercise teaching that. Could you open an issue in the C# repo?

**Comment by SleeplessByte** [comment]
Yes I'll get to that soon. 

I think the original intention of this exercise is to build something dsl-like which yeah, would be nice with a fluent interface :D

File: exercises/dot-dsl/description.md [hunk 1]
```diff
@@ -19,4 +21,11 @@ Putting this in a file `example.dot` and running `dot example.dot -T png
...
 connected by a green line on a yellow background.
 
-Create a DSL similar to the dot language.
+Write a Domain Specific Language similar to the Graphviz dot language.
+
+Our DSL is similar to the Graphviz dot language in that our DSL will be used
+to create graph data structures. However, unlike the DOT Language, our DSL will
+be an internal DSL for use only in our language. 
+
+More information about the difference between internal and external DSL can be
```
Comment (ErikSchierboom) [inline exercises/dot-dsl/description.md]:
```suggestion
More information about the difference between internal and external DSLs can be
```

Comment (aburd) [reply exercises/dot-dsl/description.md]: Ah, great catch! I'll fix that up now.

**Comment by aburd** [comment]
@coriolinus 
Thanks! @SleeplessByte gave me great advice on the direction.

**Comment by SleeplessByte** [comment]
Let's make subsequent changes in a later PR. Great work @aburd and definitely great improvement.

**Comment by aburd** [comment]
@SleeplessByte
Many thanks for your help!
